### PR TITLE
Button guidelines

### PR DESCRIPTION
**Pull request by garrettdigital**
The buttons section has been completely re-written to provide better guidance of how buttons are actually used in VS Web.

If any questions about buttons still remain unanswered then please call them out in this review so that they can be added.

### CODE REVIEW

**Comment by Woolfe-Glitch** [comment]
"As a general guideline there should only be one High emphasis button per view. Usability best practices state that the user should always have the ability to safely exit a task so it is always recommended to have a 'Cancel' button available."

It feels like this should potentially be under the next headline (Useage) instead of the general "Buttons" since it applies to how they are used.

**Comment by garrettdigital** [comment]
> "As a general guideline there should only be one High emphasis button per view. Usability best practices state that the user should always have the ability to safely exit a task so it is always recommended to have a 'Cancel' button available."
> 
> It feels like this should potentially be under the next headline (Useage) instead of the general "Buttons" since it applies to how they are used.

Good suggestion. I've re-arraigned the text a bit.

File: docs/design-language.md [hunk 1]
```diff
@@ -5,44 +5,62 @@ title: Design Language
...
+        <legend>Don't</legend>
+        <ButtonLabelsDont />
+    </fieldset>
+</div>
 
-If space is an issue, or there is no need for a description of the button to be added, there are iconographic buttons available.
+### Sizing and Spacing
 
-<div class="buttons-box"><IconButtons /></div>
+Buttons will adjust their width to the length of their content. Spacing (padding and margin) should not be manually set as they are pre-set in the Design System so that buttons can participate in the system wide density settings. Please refer to **[the Developer Center](https://developers.vertigisstudio.com/docs/web/configuration-layout-reference#presentation-attributes)** for more information on spacing.
```
Comment (eric-parsons) [inline docs/design-language.md:57]: Padding should be left alone, but margin actually is something that is set by component that is rendering the buttons. 

The general rule is that a parent component controls the positioning, flow, size (sometimes) and spacing around its child components, but should never change the look or feel of the child components themselves (i.e. no CSS that affects anything within a component's borders) beyond whatever properties are provided for this purpose, such as emphasis.

Comment (eric-parsons) [reply docs/design-language.md:57]: In the specific case of buttons, you should never set their size directly with CSS. Instead use small, medium, and large via the size prop. Same in most other cases like icons and such. The only time you would specify sizing in CSS might be say a min height on a tabs component's content area (things that generally stretch in size).

**Review by Woolfe-Glitch — APPROVED** [review]

File: docs/design-language.md (hunk 1 above)
Comment (garrettdigital) [reply docs/design-language.md]: Thanks for the clarification on the padding, margin, and sizing. I've updated the text to cover this better.

Comment (eric-parsons) [reply docs/design-language.md]: Thanks, looks good.

**Review by eric-parsons — APPROVED** [review]

**Review by jfgodin — APPROVED** [review]

**Review by P-e-t-e-r — APPROVED** [review]

**Pull request merged**